ledger/eval.go

```diff
@@ -1174,7 +1174,15 @@ func eval(ctx context.Context, l ledgerForEvaluator, blk bookkeeping.Block, vali
 		return ledgercore.StateDelta{}, err
 	}
 
-	paysetgroupsCh := loadAccounts(ctx, l, blk.Round()-1, paysetgroups, blk.BlockHeader.FeeSink, blk.ConsensusProtocol())
+	accountLoadingCtx, accountLoadingCancel := context.WithCancel(ctx)
+	paysetgroupsCh := loadAccounts(accountLoadingCtx, l, blk.Round()-1, paysetgroups, blk.BlockHeader.FeeSink, blk.ConsensusProtocol())
+	// ensure that before we exit from this method, the account loading is no longer active.
+	defer func() {
+		accountLoadingCancel()
+		// wait for the paysetgroupsCh to get closed.
```

Comment:
```suggestion
		// wait for the paysetgroupsCh to get closed and/or remove the remaining accounts from the channel.
```

Reply:
Will this stop waiting if the channel is emptied? I think it has to get closed.

Reply:
The channel being closed is an indication that the go-routine is done.. removing the accounts from the channel is not really interesting here, but rather a way to detect that the channel got closed ( which ensures that no database actions would be taken place ).

Reply:
You're right, "and/or" was wrong. I thought it was interesting to mention that if there were unprocessed accounts (i.e. if there was an error) they would be released.